src/twisted/conch/ssh/connection.py

```diff
@@ -444,10 +443,12 @@ def sendRequest(self, channel, requestType, data, wantReply=0):
         if channel.localClosed:
             return
         log.msg('sending request %r' % (requestType))
-        self.transport.sendPacket(MSG_CHANNEL_REQUEST, struct.pack('>L',
-                                    self.channelsToRemoteChannel[channel])
-                                  + common.NS(requestType)+chr(wantReply)
-                                  + data)
+        self.transport.sendPacket(
+                MSG_CHANNEL_REQUEST,
+                struct.pack('>L', self.channelsToRemoteChannel[channel])
+                + common.NS(requestType)
+                + bytes([wantReply])
```

Comment:
It seems like we could avoid the list allocation and function call here pretty easily.

```suggestion
                + b'\x01' if wantReply else b'\x00'
```

Reply:
That needs some parentheses, otherwise the operator precedence breaks things.

If we care about performance, the use of `+` over `bytes.join()` might be a bigger issue here. Also structs could be precompiled, so we wouldn't have to gamble that the built-in struct cache is large enough. But unlike those, what you suggested is a pretty small change, so I could make it if you prefer.